### PR TITLE
implement all

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/com/mate/dao/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void add(User user) {
```
Comment (AntonPanchishko) [inline src/main/java/com/mate/dao/UserDaoImpl.java:19]: (Optional) you are able to simplify your dao

File: src/main/java/com/mate/service/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.mate.dao.UserDao;
+import com.mate.model.User;
+import java.util.List;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserDao userDao;
+
+    public UserServiceImpl(UserDao userDao) {
```
Comment (AntonPanchishko) [inline src/main/java/com/mate/service/UserServiceImpl.java:12]: As i remember you should add annotation

**Review by AntonPanchishko — CHANGES_REQUESTED** [review]
Good job! small comment

File: src/main/java/com/mate/service/UserServiceImpl.java (hunk 2 above)
Comment (Resci) [reply src/main/java/com/mate/service/UserServiceImpl.java:12]: Annotation is optional

**Review by AntonPanchishko — APPROVED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**